File: client/android/UseCase/Auth/Session/RequestRecoveryUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.RequestRecoveryRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class RequestRecoveryUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: RequestRecoveryRequest): AuthSessionPayload =
        gateway.requestRecovery(request)
}

File: client/ios/UseCase/Auth/Session/RequestRecoveryUseCase.swift
public struct RequestRecoveryUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: RequestRecoveryRequest) async throws -> AuthSessionPayload {
        try await gateway.requestRecovery(request: request)
    }
}
